### DISCUSSION TITLE
About stabilizerTask's running Rate.

### DISCUSSION THREAD

**Discussion by HelloIDTC** [bitcraze/discussions#1312] [Q&A] [answered]
After tag 2018.01.1, vTaskDelayUntil(&lastWakeTime, F2T(RATE_MAIN_LOOP)) was removed from stabilizerTask. How do you ensure that this task runs at 1000Hz?
https://github.com/bitcraze/crazyflie-firmware/blob/2018.01.1/src/modules/src/stabilizer.c#L125

**Comment by gemenerik** [comment]
[This comment](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/stabilizer.c#L285) refers to it. Instead of forcing a delay, we wait for a callback from the gyro.

**Reply by HelloIDTC** [reply]
Thanks, is the gyroscope's callback time accurate? Whether time jitter occurs?

**Reply by gemenerik** [reply, marked as answer]
We configure the gyro to have an output data rate of 1000 Hz [here](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/hal/src/sensors_bmi088_bmp3xx.c#L408). I would think this is pretty constant. The Crazyflie 2.1 uses the Bosch BMI088, perhaps you can find more information about it online.

**Reply by HelloIDTC** [reply]
Thanks.
